#!/usr/bin/env python
PACKAGE = "ros_controllers"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("use_filter",    bool_t,    0, "Filter steering input", True)
gen.add("target_velocity", double_t, 0, "The target velocity",   2, 0,   10)
gen.add("Kd", double_t, 0, "Differential Gain",   0.01, 0,   10)
gen.add("Kp", double_t, 0, "Proportional Gain",    3.0, 0,   10)
gen.add("Ki", double_t, 0, "Integrator Gain",    0.0, 0,   10)
gen.add("lag_compensation_time", double_t, 0, "lag_compensation_time",    0.2, 0,   10)

gen.add("K_torque_curv", double_t, 0, "K_torque_curv",   0.32, 0,   10)
gen.add("mean_curv_dist", int_t, 0, "mean_curv_dist (in number of track indices)",   0, 70,   200)
gen.add("look_ahead_dist", double_t, 0, "look_ahead_dist",   0, 0.15,   10)

exit(gen.generate(PACKAGE, "ff_fb_config", "ff_fb"))
